### DISCUSSION TITLE
Failure while downloading  reranker model

### DISCUSSION THREAD

**Discussion by Pareek-Yash** [open-webui/open-webui#5591] [General]
# Bug Report

## Description
Failure to download the reranker model from the UI, UI hangs and healthcheck fails. Need to rebuild the container every time.
```
"GET /ws/socket.io/?EIO=4&transport=polling&t=P3UgWz5&sid=bsurF5q0lo1p8SNnAAEd HTTP/1.0" 200 OK
INFO  [apps.rag.main] Updating reranking model:  to BAAI/bge-reranker-v2-m3
INFO  [config] Saving 'RAG_RERANKING_MODEL' to config.json
ERROR [apps.rag.utils] Cannot determine model snapshot path: Cannot find an appropriate cached snapshot folder for the specified revision on the local disk and outgoing traffic has been disabled. To enable repo look-ups and downloads online, pass 'local_files_only=False' as input.
Traceback (most recent call last):
  File "/app/backend/apps/rag/utils.py", line 353, in get_model_path
    model_repo_path = snapshot_download(**snapshot_kwargs)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/huggingface_hub/utils/_validators.py", line 114, in _inner_fn
    return fn(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/huggingface_hub/_snapshot_download.py", line 220, in snapshot_download
```
**Bug Summary:**
[Provide a brief but clear summary of the bug]

**Steps to Reproduce:**
From the webui/admin panel/documents click hybrid search and add `BAAI/bge-reranker-v2-m3` in the field, and click download icon. It keeps downloading and the server fails.

**Expected Behavior:**
Should download model and ready for queries.

**Actual Behavior:**
Server crashes

## Environment

- **Open WebUI Version:** [e.g., 0.1.120]
- **Ollama (if applicable):** [e.g., 0.1.30, 0.1.32-rc1]

- **Operating System:** Ubuntu 22.04
- **Browser (if applicable):** [e.g., Chrome 100.0, Firefox 98.0]

## Reproduction Details

**Confirmation:**

- [x] I have read and followed all the instructions provided in the README.md.
- [x] I am on the latest version of both Open WebUI and Ollama.
- [] I have included the browser console logs.
- [x] I have included the Docker container logs.

## Logs and Screenshots

**Browser Console Logs:**
[Include relevant browser console logs, if applicable]

**Docker Container Logs:**
```
"GET /ws/socket.io/?EIO=4&transport=polling&t=P3UgWz5&sid=bsurF5q0lo1p8SNnAAEd HTTP/1.0" 200 OK
INFO  [apps.rag.main] Updating reranking model:  to BAAI/bge-reranker-v2-m3
INFO  [config] Saving 'RAG_RERANKING_MODEL' to config.json
ERROR [apps.rag.utils] Cannot determine model snapshot path: Cannot find an appropriate cached snapshot folder for the specified revision on the local disk and outgoing traffic has been disabled. To enable repo look-ups and downloads online, pass 'local_files_only=False' as input.
Traceback (most recent call last):
  File "/app/backend/apps/rag/utils.py", line 353, in get_model_path
    model_repo_path = snapshot_download(**snapshot_kwargs)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/huggingface_hub/utils/_validators.py", line 114, in _inner_fn
    return fn(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/huggingface_hub/_snapshot_download.py", line 220, in snapshot_download
```

**Screenshots (if applicable):**
[Attach any relevant screenshots to help illustrate the issue]

## Installation Method

Docker installation

## Additional Information

Added the following suggestions but still crashes: `RAG_EMBEDDING_MODEL_AUTO_UPDATE=False`

and

`RAG_EMBEDDING_MODEL_TRUST_REMOTE_CODE = True`

## Note

If the bug report is incomplete or does not follow the provided instructions, it may not be addressed. Please ensure that you have followed the steps outlined in the README.md and troubleshooting.md documents, and provide all necessary information for us to reproduce and address the issue. Thank you!

**Comment by luisbrandao** [comment]
Try with:

```
RAG_EMBEDDING_MODEL_AUTO_UPDATE=True
RAG_RERANKING_MODEL_AUTO_UPDATE=True
RAG_EMBEDDING_MODEL_TRUST_REMOTE_CODE=True
RAG_RERANKING_MODEL_TRUST_REMOTE_CODE=True
```
As env. Worked for me.

**Comment by thomashkolbe** [comment]
Hello, I have the same problem, and setting the four environment variables as suggested above during the startup of the Docker container does not help. Keeps giving the same error message.

**Comment by damajor** [comment]
I had some non sense issue, and the root cause was ipv6 inside the container. Maybe it worth a try to disable it.

**Comment by D1yt** [comment]
Mind reporting that? I have issues with every reranker other than v2_m3

**Comment by damajor** [comment]
> Mind reporting that? I have issues with every reranker other than v2_m3
> […](#)
> On Sat, Jul 27, 2024, 19:36 -_- ***@***.***> wrote: I had some non sense issue, and the root cause was ipv6 inside the container. Maybe it worth a try to disable it.

My issue was related to web search.

**Comment by DennisWangCW** [comment]
same error encountered, waiting for a resolution

**Comment by sir3mat** [comment]
Same problem here but I think is due to the fact that BAAI/bge-reranker-v2-m3 is not supporter by sentence transformer. 
For example, with this reranker https://huggingface.co/mixedbread-ai/mxbai-rerank-large-v1 it works well (also web search)

**Comment by D1yt** [comment]
I think I got BAAI/bge-reranker-v2-m3 running.
Container breaks when trying v2-gemma though.

I used the env vars:
```
RAG_EMBEDDING_MODEL_AUTO_UPDATE=True
RAG_RERANKING_MODEL_AUTO_UPDATE=True
RAG_EMBEDDING_MODEL_TRUST_REMOTE_CODE=True
RAG_RERANKING_MODEL_TRUST_REMOTE_CODE=True
```

But I didn't change all the parameters in one go, I left the reranker model blank, saving everything else first. And in the last step I enter the reranker model and give it some time to download all files (monitor the logs).

![Screenshot_20240731-100826](https://github.com/user-attachments/assets/b6e8fa76-9269-4824-8fde-7939a935240c)


Important to note, that I am on a docker-compose setup with ollama always being up-to-date.
The bundled installation had my ollama running some versions behind.

**Comment by sir3mat** [comment]
I think that BAAI/bge-reranker-v2-m3 is installed in the docker container of open web ui and didn't use the ollama instance.
 If you check the logs of the openwebui container, immediately after a request, it logs any information regarding sentence transformer unable to load the last two layer of v2_m3 (unable to load weights and biases)? the reranker is a xlmroberta with finetuning and cross encoder if i remember correctly and I think that sentence transformer is not able to load this specific architecture (with the addition of the cross encoder)
 
 For v2-gemma i don't have any experience but i suppose it the same problem.
 
 Did you have the models downloaded in the huhhigface cache folder of the openwebui container ( path: /apps/backend/data/cache/embeddings/model or something like this)?

**Comment by D1yt** [comment]
![Screenshot_20240731-190431](https://github.com/user-attachments/assets/d24edc6f-a3d4-4e57-9938-46a4b6d4a387)

My logs (with no word about the reranker):
[open-webui_logs(1).txt](https://github.com/user-attachments/files/16468737/open-webui_logs.1.txt)
[ollama_logs.txt](https://github.com/user-attachments/files/16468738/ollama_logs.txt)

**Comment by OM-EL** [comment]
any one know how we can download the models manually if we want  ?

**Comment by sir3mat** [comment]
I think that From the UI, paste the model name and click the icon with the arrow ⬇️
If you want to do it manually I suppose you can enter the terminal of docker container and use hugging face cli

**Comment by D1yt** [comment]
> ![Screenshot_20240731-190431](https://private-user-images.githubusercontent.com/10422925/353921503-d24edc6f-a3d4-4e57-9938-46a4b6d4a387.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjI1OTIyMzcsIm5iZiI6MTcyMjU5MTkzNywicGF0aCI6Ii8xMDQyMjkyNS8zNTM5MjE1MDMtZDI0ZWRjNmYtYTNkNC00ZTU3LTk5MzgtNDZhNGI2ZDRhMzg3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODAyVDA5NDUzN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQ3ZGI5YmI5M2YxOGEzYjljZmU0MWE2ZWY5MDNjYTU1ODY1YTEyMjU2MGFjNTNlODg3Y2JmYjUzMjg3N2MyZTAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.iVVkTgg8nuyPfOJgNK3d7Rtyh6MUpWKtfHwvLWxxMAo)
> 
> My logs (with no word about the reranker): [open-webui_logs(1).txt](https://github.com/user-attachments/files/16468737/open-webui_logs.1.txt) [ollama_logs.txt](https://github.com/user-attachments/files/16468738/ollama_logs.txt)

Uploaded my logs. Weirdly enough, my AI models tend to forget pre-uploaded files (knowledge base) and just answer like they aren't present.

**Comment by D1yt** [comment]
> > ![Screenshot_20240731-190431](https://private-user-images.githubusercontent.com/10422925/353921503-d24edc6f-a3d4-4e57-9938-46a4b6d4a387.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjI1OTIyMzcsIm5iZiI6MTcyMjU5MTkzNywicGF0aCI6Ii8xMDQyMjkyNS8zNTM5MjE1MDMtZDI0ZWRjNmYtYTNkNC00ZTU3LTk5MzgtNDZhNGI2ZDRhMzg3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODAyVDA5NDUzN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQ3ZGI5YmI5M2YxOGEzYjljZmU0MWE2ZWY5MDNjYTU1ODY1YTEyMjU2MGFjNTNlODg3Y2JmYjUzMjg3N2MyZTAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.iVVkTgg8nuyPfOJgNK3d7Rtyh6MUpWKtfHwvLWxxMAo)
> > My logs (with no word about the reranker): [open-webui_logs(1).txt](https://github.com/user-attachments/files/16468737/open-webui_logs.1.txt) [ollama_logs.txt](https://github.com/user-attachments/files/16468738/ollama_logs.txt)
> 
> Uploaded my logs. Weirdly enough, my AI models tend to forget pre-uploaded files (knowledge base) and just answer like they aren't present.

Version 0.3.11 seems to have fixed my models from forgetting the knowledge base:
- 🔧 TAGS HANDLING IN DOCUMENT UPLOAD
Tags are now properly sent to the upload document handler, resolving issues with missing metadata.
- 📂 STATIC PATH RESOLUTION IN PDF GENERATION
Fixed static paths that adjust dynamically to prevent issues across various environments.

> Same problem here but I think is due to the fact that BAAI/bge-reranker-v2-m3 is not supporter by sentence transformer. For example, with this reranker https://huggingface.co/mixedbread-ai/mxbai-rerank-large-v1 it works well (also web search)

This shouldn't be the case as this reranking model is explicitly listed as an example.

**Comment by dreamworldsky** [comment]
I faced the problem too with different reason：
Fetching 15 files:   0%|          | 0/15 [00:00<?, ?it/s]
Fetching 15 files:   7%|▋         | 1/15 [00:00<00:10,  1.33it/s]
Fetching 15 files:  20%|██        | 3/15 [00:00<00:03,  3.97it/s]
Cannot determine model snapshot path: (MaxRetryError("HTTPSConnectionPool(host='cdn-lfs-us-1.hf.co', port=443): Max retries exceeded with ur：XXX [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self-signed certificate in certificate chain
my open-webui version：0.5.10
my compute access hugging face throught an Proxy
Does anyone know how to fix it？
